### DISCUSSION TITLE
Build a Wordpress plugin for PyScript

### DISCUSSION THREAD

**Discussion by jezdez** [pyscript/pyscript#1167] [Proposals]
Wordpress is a [major CMS](https://blog.hubspot.com/website/wordpress-stats) (43% of all websites, 65% marketshare of websites with a CMS) and provides a rich plugin API.

To be able to reach more people and lower the barrier of entry for regular users, we should build and publish a plugin for Wordpress that:

- simplifies adding PyScript to Wordpress sites (auto-add the script and stylesheet tags)
- provides [custom blocks](https://developer.wordpress.org/block-editor/) for their new visual editor "[Gutenberg](https://wordpress.org/gutenberg/)" (a custom block per built-in PyScript tag?)
- do regular releases of the plugin to the Wordpress plugin index following our (future) release process (automatically?)

There is [documentation for creating a plugin](https://developer.wordpress.org/block-editor/how-to-guides/) that adds new blocks and many more information on the wordpress site.

**Comment by rus19023** [comment]
This is a great idea! I have been wanting to start learning to build plugins for a while now, I think now is the time! I will help!

**Comment by bwinkers** [comment]
If this moves forwards ping me, I'll follow and may be able to contribute.
